src/Extensions/Logging/Internal/Replayable/ReplayibleActivityProvider.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.Omex.Extensions.TimedScopes;
+
+namespace Microsoft.Omex.Extensions.Logging.Replayable
+{
+	internal class ReplayibleActivityProvider : IActivityProvider
+	{
+		public ReplayibleActivityProvider(IOptions<OmexLoggingOptions> options) =>
+			m_replayLogsInCaseOfError = options.Value.ReplayLogsInCaseOfError;
+
+
+		public Activity Create(string operationName) =>
```

Comment:
Validate

Reply:
Switched to use `TimedScopeDescription` struct that will have validation inside it